src/components/pages/__tests__/dashboard.vendor.[vendorId].test.tsx

```diff
@@ -0,0 +1,78 @@
+import { render, screen, waitFor } from '@testing-library/react'
+
+import { withWrappers } from '@/lib/testing/utils.tsx'
+
+import VendorDashboardPage from '../dashboard/vendor/[vendorId].tsx'
+
+const { mockUseParams } = vi.hoisted(() => ({
+  mockUseParams: vi.fn(() => ({ vendorId: '2508' })),
+}))
+
+vi.mock('@tanstack/react-router', async () => {
+  const actual = await vi.importActual('@tanstack/react-router')
+
+  return {
+    ...actual,
+    useParams: mockUseParams,
+  }
+})
+
+describe('<DashboardPage />', () => {
+  it('should handle null response in providesTags', async () => {
+    mockUseParams.mockReturnValue({ vendorId: '-1' })
+
+    const { container } = render(
+      withWrappers(<VendorDashboardPage />, { withRoot: true }),
+    )
+
+    await waitFor(async () => {
+      screen.debug()
```

Comment:
Please delete this, this will leave a lot of unnecessary logs within the testing terminal